lib/engine/game/g_18_co.rb

```diff
@@ -185,6 +185,23 @@ def action_processed(action)
         end
       end
 
+      def check_distance(route, visits)
+        super
+
+        distance = route.train.distance
+
+        return if distance.is_a?(Numeric)
+
+        cities_allowed = distance.select { |d| d['nodes'].include?('city') }.first['pay']
```

Comment:
distance.find { ...}[0]['pay']

Reply:
Updated

Reply:
`distance.select { ... }` returns an array that is filtered down from `distance`, while `distance.find { ...}` returns just the first element from `distance` that matches the filter.

I think instead of `distance.find { ...}[0]['pay']`, you just want `distance.find { ...}['pay']`

Reply:
Thanks @michaeljb - yeah, I have a bad habit of that. I updated to your fix and actually tested this time.